Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -181,5 +217,51 @@ public DBResult<Communication> Delete(Communication communication, bool commit =
             this.logger.LogDebug($"Finished deleting Communication in DB");
             return result;
         }
+
+        private static bool AllowUpdateCurrentItem(List<Communication> activeBannerCommunication, Communication communication)
+        {
+            return activeBannerCommunication?.Count == 1 &&
+                     activeBannerCommunication.First().Id == communication.Id;
+        }
+
+        private static bool IsValidCommunication(List<Communication> activeBannerCommunication, Communication communication)
+        {
+            return activeBannerCommunication
+                              .Where(c => (c.ExpiryDateTime < communication.ExpiryDateTime &&
+                                  communication.EffectiveDateTime > c.ExpiryDateTime)
+                                  || (communication.EffectiveDateTime < c.EffectiveDateTime &&
+                                  communication.ExpiryDateTime < c.EffectiveDateTime))
+                              .AsEnumerable().Any();
+        }
+
+        private bool IsCommunicationValidCommit(Communication communication)
```

Comment:
Can we investigate to see if we can catch an exception related to a constraint violation and return a custom message instead of doing this query.

Reply:
@sslaws @MikeLyttle , Based on the article. We already used the correct exception which is ### DBUpdateException [https://entityframework.net/knowledge-base/31515776/how-can-i-catch-uniquekey-violation-exceptions-with-ef6-and-sql-server-](url) 

Then if we really want to go on the specific detail of getting the unique constraint error, we can use the sql exception number. But I think it is not safe to use the number. I think just checking  the message is enough for this case.

Reply:
@sslaws @MikeLyttle . And for concurrency issue. Have we try to use "[Timestamp]" ?[https://www.learnentityframeworkcore5.com/handling-concurrency-in-ef-core](url)